components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +133,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.having(FunctionType.TO_BOOLEAN, column, OperatorType.EQUALS_TO, "TRUE");
```

Comment:
This seems to be working. What do you think about it (@LosD, @kaspersorensen)? I think something like "strlen(column) > 0" would be much more readable but I did not find that function anywhere.